From 0851c169124d6ffdb5fd0c0c9b4cec8569b799a4 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 16 Nov 2020 17:52:17 +0200
Subject: [PATCH] [SW-12755] module_lkd: introduce enable() callback

Introduce module_lkd_enable() callback as a preparation for adding
dynamic security.
In dynamic security setup() and enable() callbacks will differ.

Change-Id: I2e39464aee40f4acd22dcaa2d5365e51a0aacefe
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/cmd_module.h b/common/cmd_module.h
index 91ceb1d..c8efd26 100644
--- a/common/cmd_module.h
+++ b/common/cmd_module.h
@@ -86,6 +86,7 @@
 /* LKD */
 void module_lkd_setup(void);
 enum cmd_check_res module_lkd_check(void);
+void module_lkd_enable(void);
 void module_lkd_disable(void);
 void module_lkd_run(void);
 void module_lkd_run_noret(void);
diff --git a/preboot/module_lkd.c b/preboot/module_lkd.c
index 855bef7..6e6c2ae 100644
--- a/preboot/module_lkd.c
+++ b/preboot/module_lkd.c
@@ -20,11 +20,16 @@
 
 static struct lkd_status sts;
 
-void module_lkd_setup(void)
+void module_lkd_enable(void)
 {
 	cpu_boot_sts_set(CPU_BOOT_STATUS_WAITING_FOR_BOOT_FIT);
 }
 
+void module_lkd_setup(void)
+{
+	module_lkd_enable();
+}
+
 enum cmd_check_res module_lkd_check(void)
 {
 	enum cmd_check_res res;
diff --git a/preboot/preboot.c b/preboot/preboot.c
index fd4412b..78f38a9 100644
--- a/preboot/preboot.c
+++ b/preboot/preboot.c
@@ -47,7 +47,7 @@
 		.check = module_lkd_check,
 		.disable = module_lkd_disable,
 		.run = module_lkd_run,
-		.enable = module_lkd_setup,
+		.enable = module_lkd_enable,
 		.run_fast = noop,
 		.teardown = module_lkd_disable,
 		.run_noreturn = module_lkd_run_noret,
